### PR TITLE
[batch] JVM container logs streamed with fluentd

### PR DESCRIPTION
**Pull request by jigold**
The goal of this PR is to have all of the JVM container logs available where all the worker logs are. I tagged the entries with "worker.log" so they show up with the other worker log entries. However, it's plain text with no timestamp. We can improve the formatting as a separate project. Notice the two entries with "*" on the left instead of the normal "I".

The design choice I made is to have the JVM containers write to a location that is static. We cannot easily change the fluentd configuration dynamically. It requires restarting the daemon which takes 1.5 seconds. Furthermore, the configuration for fluentd is on /etc/ on the host which the batch worker container cannot access. Hence, why I took the approach of specifying it in the startup script at known locations. 

Before we merge this, I'd like to confirm that (a) we want these logs and (b) they don't contain any secrets.
Screenshot 2023-06-16 at 4 06 43 PM

### CODE REVIEW

**Comment by jigold** [comment]
cc: @iris-garden This is the feature we discussed last week that might help debugging Query.

**Review by daniel-goldstein — CHANGES_REQUESTED** [review]
Nice, just a couple small things.

File: batch/batch/cloud/gcp/driver/create_instance.py [hunk 1]
```diff
@@ -75,6 +75,32 @@ def create_vm_config(
...
+<source>
+@type tail
+<parse>
+    # 'none' indicates the log is unstructured (text).
+    @type none
+</parse>
+path {log_path}
+pos_file /var/lib/google-fluentd/pos/jvm-{idx}.pos
+read_from_head true
+tag worker.log
```
[the remark this answers is not in the document]
Comment (jigold) [reply batch/batch/cloud/gcp/driver/create_instance.py:96]: Do you prefer `jvm-{idx}.log` or `jvm.log`?

[the remark this answers is not in the document]
Comment (jigold) [reply batch/batch/cloud/gcp/driver/create_instance.py]: I just kept it as `jvm-{idx}.log`

File: batch/batch/cloud/gcp/driver/create_instance.py [hunk 2]
```diff
@@ -235,6 +263,11 @@ def scheduling() -> dict:
 
 sudo mkdir -p /etc/netns
 
+mkdir -p /batch/jvm-container-logs/
+{jvm_touch_command}
+
+sudo service google-fluentd restart
```
[the remark this answers is not in the document]
Comment (jigold) [reply batch/batch/cloud/gcp/driver/create_instance.py:269]: These files need to be touched after the XFS file system and all is created. I moved the rest of the fluentd configuration code down so it's all together.

**Review by daniel-goldstein — APPROVED** [review]

**Pull request merged**